doc/VTGateV3Features.md

```diff
@@ -0,0 +1,77 @@
+
+### Consistency
+Once you add multiple indexes to tables, it's possible that the application could make inconstent requests. V3 makes sure that none of the specified constraints are broken. For example, if a table had both a basic sharding key and a hashed sharding key, it will enforce the rule that the hash of the basic sharding key matches that of the hashed sharding key.
+
+Some of the changes require updates to be performed across multiple databases. For example, inserting a row into a table that has a cross-shard key requires an additional row to be inserted into the lookup table. This results in distributed transactions. Currently, this is a best effort update. It is possible that partial commits happen if databases fail in the middle of a distributed commit. *We will soon develop support for 2PC to overcome this limitation.*
+
+### Query diversity
+V3 does not support the full SQL feature set. The current implementation supports simple queries:
+* Single table DML statements: This is a vitess-wide restriction where you can affect only one table and one sharding key per statement. *This restriction may be limited in the future.*
+* Single table SELECT statements:
+  * All constructs allowed if the statement targets only a single sharding key
+  * Aggregation and sorting not allowed if the statement targets more than one sharding key. Selects are allowed to target multiple sharding keys as long as the results from individual shards can be simply combined together to form the final result.
+
+In the immediate future, we plan to add support for the following constructs:
+* Joins that can be served by sending the query to a single shard.
+* Joins that can be served by sending the query to multiple shards, and trivially combined to form the final result.
+* Cross-shard joins that can be served through a simple nested lookup.
+* Sorting that can be trivially merged from the results of multiple shards (merge-sort).
+* Aggregations (and grouping) that can be trivially combined from multiple shards.
+* A combination of the above constructs as long as the results remain trivially combinable.
+
+SQL is a very powerful language. You can build queries that can result in large amount of work and memory consumption involving big intermediate results. Such constructs where the scope of work is open-ended will not be immediately supported. In such cases, it's recommended that you use map-reduce techniques for which there is a separate API. *We'll consider building such on-the-fly map-reducers in the future.*
+
+## The vschema editor
+The above features require metadata like sharding key, cross-shard indexes, etc. to be configured and stored in some place. This is known as the vschema. *We'll be building a vschema editor and wizard that will allow you to easily view and modify this metadata.*
```

Comment:
I'd rephrase as: "metadata like configuration of sharding key and cross-shard indexes" to make it clear that it's the configuration information that is being stored in the vschema not the entities themselves.